pkg/controller/gkenetworkparamset/gnpcontroller_validations.go

```diff
@@ -217,6 +217,21 @@ func crossValidateNetworkAndGnp(network *networkv1.Network, params *networkv1.GK
 				ErrorMessage: "L3 type network requires secondary range to be specified in params",
 			}
 		}
+
+		// if the "default" params has new pod ranges added from Multi Pod CIDR,
+		// marks the Network not ready. Only apply to new pod ranges to limit the
+		// reconciliation of GNP and Network object
+		if network.Name == networkv1.DefaultPodNetworkName &&
+			params.Name == networkv1.DefaultPodNetworkName &&
+			originalParams.Status.PodCIDRs != nil &&
+			len(originalParams.Status.PodCIDRs.CIDRBlocks) > 0 &&
+			!sameStringSlice(params.Status.PodCIDRs.CIDRBlocks, originalParams.Status.PodCIDRs.CIDRBlocks) {
+			return &gnpNetworkCrossValidation{
+				IsValid:      false,
+				ErrorReason:  networkv1.GNPParamsNotReady,
+				ErrorMessage: "New Pod ranges in default VPC requires network CIDRs update",
```

Comment:
just for learning, who is going to make these updates?

Reply:
GNP controller should change the Network condition, this will trigger NCM to update Network object. Moved to another place, please review again.